@claude CI failed on HEAD `d7272ff5`. Please fix the failures below and push to this branch.

## Failed checks
- **github-actions** / `main (3.11)` — https://github.com/MervinPraison/PraisonAI/actions/runs/32363476643/job/96410291402

## Failures (extracted)
1. `tests/unit/compute/test_docker_capture.py::test_registry_records_definition_label` — `IndexError: list index out of range`
   - Job: `main (3.11)`
2. `tests/unit/compute/test_docker_capture.py::test_concurrent_captures_preserve_each_other` — `assert 0 == 2`
   - Job: `main (3.11)`
3. `tests/unit/compute/test_docker_capture.py::test_differing_env_values_do_not_share_capture` — `IndexError: list index out of range`
   - Job: `main (3.11)`
4. `tests/unit/compute/test_docker_capture.py::test_touch_capture_updates_last_used` — `IndexError: list index out of range`
   - Job: `main (3.11)`
5. `tests/unit/compute/test_docker_capture.py::test_registry_gc_prune` — `assert 0 == 1`
   - Job: `main (3.11)`
6. `tests/unit/compute/test_docker_capture.py::test_refresh_runs_only_on_capture_start` — `AssertionError: assert [['sh', '-c', 'make setup']] == [['sh', '-c',...nstall -e .']]`
   - Job: `main (3.11)`
7. `tests/unit/compute/test_docker_capture.py::test_definition_change_invalidates` — `IndexError: list index out of range`
   - Job: `main (3.11)`
8. `tests/unit/compute/test_docker_capture.py::test_docker_capture_and_reuse` — `assert 0 == 1`
   - Job: `main (3.11)`

## Critical review first
Before changing code or tests, decide **which side is wrong**:
1. **Legitimate feature change** — the PR intent is correct but implementation or tests need updating. Preserve SDK guarantees; update tests only when behaviour intentionally changed and document why.
2. **Regression / bug in this PR** — the failure exposes a real breakage introduced here. Fix the implementation; **do not weaken, skip, or delete tests** just to go green.
3. **Pre-existing flake or unrelated failure** — say so explicitly; prefer fixing the root cause over masking it.

**SDK guardrails (AGENTS.md):**
- Agent-callable tools → PraisonAI-Tools; lifecycle plugins → PraisonAI-Plugins; sandbox backends → PraisonAI-Plugins (`praisonai.sandbox`) — not praisonaiagents/.
- Do not disturb core SDK contracts to accommodate a wrapper/feature change.
- Tests must continue to guard backward compatibility and hot-path behaviour — passing CI by lowering test standards is not acceptable.
- If the feature does not genuinely add SDK value, recommend reverting or narrowing scope instead of patching around failures.

## What to do
1. State your verdict: **legitimate fix**, **regression fix**, or **needs human review** — and why (1–3 sentences).
2. Fix root cause with **minimal changes**; never bloat the Agent class with extra params.
3. Run failing tests locally, e.g. `pytest tests/unit/compute/test_docker_capture.py::test_registry_records_definition_label -q`, plus any related SDK tests touched by the PR.
4. Push to this branch and comment: files changed, review verdict, and why tests still protect SDK behaviour.

--- a/src/praisonai/tests/unit/compute/test_docker_capture.py
+++ b/src/praisonai/tests/unit/compute/test_docker_capture.py
@@ -48,7 +48,11 @@ class _FakeContainer:
             return 0, (b"", b"")
         return 0, b""
 
-    def commit(self, repository=None, tag=None):
+    def commit(self, repository=None, tag=None, changes=None, **kwargs):
+        # Mirror docker-py's Container.commit signature: extra Dockerfile
+        # instructions arrive via ``changes`` (used to scrub secret ENV lines
+        # on capture). The fake records the ref just as before; the presence of
+        # ``changes`` must not turn a successful commit into an ephemeral one.
         if self._client.commit_fails:
             raise RuntimeError("commit boom")
         self._client.images._store.add(f"{repository}:{tag}")

